### PR TITLE
Fontconfig windows workaround v2

### PR DESCRIPTION
**Pull request by saurtron**
Better version reusing previous path but with mangled xml.

### CODE REVIEW

File: rts/Rendering/Fonts/CFontTexture.cpp [hunk 1]
```diff
@@ -150,23 +150,25 @@ class FtLibraryHandler {
 			// and system fonts included. also linux actually has system config files that can be used by fontconfig.
 
 			#ifdef _WIN32
-			// fontconfig will resolve the special keys here.
-			static constexpr const char* configFmt = R"(<fontconfig><dir>WINDOWSFONTDIR</dir><cachedir>fontcache</cachedir><cachedir>LOCAL_APPDATA_FONTCONFIG_CACHE</cachedir></fontconfig>)";
+			std::array<char, 4096> osFontsDir;
+			ExpandEnvironmentStrings("%WINDIR%\\fonts", osFontsDir.data(), osFontsDir.size()); // expands %HOME% etc.
+			static constexpr const char* configFmt = R"(<fontconfig><dir>%s</dir><cachedir>fontcache</cachedir></fontconfig>)";
+			std::string configFmtVar = fmt::sprintf(configFmt, osFontsDir.data());
```
Comment (lhog) [inline rts/Rendering/Fonts/CFontTexture.cpp:156]: Can be `const` too for the sake of uniformity, but not really critical.

File: rts/Rendering/Fonts/CFontTexture.cpp [hunk 2]
```diff
@@ -150,23 +150,25 @@ class FtLibraryHandler {
 			// and system fonts included. also linux actually has system config files that can be used by fontconfig.
 
 			#ifdef _WIN32
-			// fontconfig will resolve the special keys here.
-			static constexpr const char* configFmt = R"(<fontconfig><dir>WINDOWSFONTDIR</dir><cachedir>fontcache</cachedir><cachedir>LOCAL_APPDATA_FONTCONFIG_CACHE</cachedir></fontconfig>)";
+			std::array<char, 4096> osFontsDir;
+			ExpandEnvironmentStrings("%WINDIR%\\fonts", osFontsDir.data(), osFontsDir.size()); // expands %HOME% etc.
```
Comment (sprunk) [inline rts/Rendering/Fonts/CFontTexture.cpp:154]: In theory the constant 4096 is incorrect, maybe the below is overcomplicating tho?
```suggestion
			static constexpr auto winFontPath = "%WINDIR%\\fonts";
			const int neededSize = ExpandEnvironmentStrings(winFontPath, nullptr, 0);
			std::vector <char> osFontsDir (neededSize);
			ExpandEnvironmentStrings(winFontPath, osFontsDir.data(), osFontsDir.size());
```

Comment (lhog) [reply rts/Rendering/Fonts/CFontTexture.cpp:154]: It's only called once per engine start so we can afford extra `vector` allocation. Not a big deal.

Comment (saurtron) [reply rts/Rendering/Fonts/CFontTexture.cpp]: yeah it does look overcomplicated but ok I commited that, maybe later on that can be put into some sort of utility function somewhere, I see ExpandEnvironmentStrings being used elsewhere with a much bigger buffer.

**Comment by lhog** [comment]
LGTM

**Pull request merged**